src/HTTP.jl

```diff
@@ -626,12 +636,54 @@ function stack(;redirect=true,
     (verbose >= 3 ||
      DEBUG_LEVEL[] >= 3   ? DebugLayer                : NoLayer){
     (readtimeout > 0      ? TimeoutLayer              : NoLayer){
-                            StreamLayer
+                            StreamLayer{Union{}}
     }}}}}}}}}}}}
 end
 
-include("download.jl")
 
+"""
+    next(::Type{S}) where {T, S<:Layer{T}}
+
+Return the next layer in the stack
+"""
+function next(::Type{S}) where {T, S<:Layer{T}}
+    return T
+end
+
+"""
+    head(::Type{T}) where T <: Layer
+
+Get the name of the type
```

Comment:
>  Get the name of the type

That is **not** what this does.

```
julia> struct Foo{X} end

julia> nameof(Foo{Int})
:Foo

julia> Foo{Int}.name
Foo

julia> Foo{Int}.name.wrapper
Foo

julia> Foo{Int}.name |> typeof
Core.TypeName

julia> Foo{Int}.name.wrapper |> typeof
UnionAll

julia> Foo{Int}.name.wrapper == Foo{T} where T
true
```

I think this only applies to parametric types?
That should be made clear in docstring also